ext/ManoptJuMPExt.jl

```diff
@@ -304,36 +327,20 @@ function MOI.optimize!(model::Optimizer)
             model.variable_primal_start[i]
         end for i in eachindex(model.variable_primal_start)
     ]
-    backend = MOI.Nonlinear.SparseReverseMode()
-    vars = [MOI.VariableIndex(i) for i in eachindex(model.variable_primal_start)]
-    evaluator = MOI.Nonlinear.Evaluator(model.nlp_model, backend, vars)
-    MOI.initialize(evaluator, [:Grad])
-    function eval_f_cb(M, x)
-        if model.sense == MOI.FEASIBILITY_SENSE
-            return 0.0
-        end
-        obj = MOI.eval_objective(evaluator, JuMP.vectorize(x, _shape(model.manifold)))
-        if model.sense == MOI.MAX_SENSE
-            obj = -obj
+    objective = model.objective
+    if isnothing(objective) || model.sense == MOI.FEASIBILITY_SENSE
+        eval_f_cb(_, _) = 0.0
+        function eval_grad_f_cb(M, X)
+            x = JuMP.vectorize(X, _shape(model.manifold))
+            grad_f = zeros(length(x))
+            reshaped_grad_f = JuMP.reshape_vector(grad_f, _shape(model.manifold))
+            return ManifoldDiff.riemannian_gradient(model.manifold, X, reshaped_grad_f)
         end
-        return obj
-    end
-    function eval_grad_f_cb(M, X)
-        x = JuMP.vectorize(X, _shape(model.manifold))
-        grad_f = zeros(length(x))
-        if model.sense == MOI.FEASIBILITY_SENSE
-            grad_f .= zero(eltype(grad_f))
-        else
-            MOI.eval_objective_gradient(evaluator, grad_f, x)
-        end
-        if model.sense == MOI.MAX_SENSE
-            LinearAlgebra.rmul!(grad_f, -1)
-        end
-        reshaped_grad_f = JuMP.reshape_vector(grad_f, _shape(model.manifold))
-        return ManifoldDiff.riemannian_gradient(model.manifold, X, reshaped_grad_f)
+        objective = Manopt.ManifoldGradientObjective(eval_f_cb, eval_grad_f_cb)
```

Comment:
I did this at the moment as it's equivalent to what we had before this PR but that's probably a bit convoluted ^^

Reply:
well if the objective is already nothing then doing a ManifoldDiffGrad of that is not so reasonable ;)

Reply:
What is a ManifoldDiffGrad ?

Reply:
Sorry just me being lazy to not write the full `ManifoldDiff.riemannian_gradient(...)`.

Reply:
`riemannian_gradient` is in the removed lines of the diff

Reply:
Yes, probably, `zero(X)` would work

Reply:
That is `X` here? The tangent vector (gradient)? Then we have `zero_vector(M,p)` for that. But no need to do a gradient, when you use the `Manifold)CostObjective` (which does not have a gradient)

Reply:
But then we'll have to use a zero-order method instead of a first-order one right ?

Reply:
Sure, but most of that code is magic to me that comes from MOI here, so I have not yet understood why the objective might suddenly be nothing.

And with a constant cost neither zeroth nor first order methods make sense either.

So I am not yet sure what all this case is about.

Reply:
This happens if the user don't use `@objective`. The way you can interpret this is that the solver is free to return any point of the manifold